src/Microsoft.DotNet.Interactive.Http.Tests/HttpKernelTests.cs

```diff
@@ -336,15 +442,832 @@ public async Task url_in_embedded_expression_is_valid()
         using var _ = new AssertionScope();
 
         var code = """
-            @hostname=httpbin.org
-            @host=https://{{hostname}}
-            POST {{host}}/anything
+            POST https://api.example.com/comments
+            
+            {
+                "updated_at" : "{{$timestamp}}"
+            }
             """;
 
         var result = await kernel.SendAsync(new SubmitCode(code));
         result.Events.Should().NotContainErrors();
 
-        request.RequestUri.AbsoluteUri.Should().Be("https://httpbin.org/anything");
+        var bodyAsString = await request.Content.ReadAsStringAsync();
+        var unixValueSubstring = bodyAsString.Split(":").Last().Trim().Substring(1);
+        var unixValueString = unixValueSubstring.Substring(0, unixValueSubstring.IndexOf("\""));
+        var unixValue = DateTimeOffset.FromUnixTimeSeconds(long.Parse(unixValueString));
+        unixValue.Should().BeCloseTo(DateTimeOffset.UtcNow, TimeSpan.FromSeconds(10));
+    }
+
+    [Theory]
+    [InlineData("-1")]
+    [InlineData("+1")]
+    [InlineData("0")]
+    [InlineData("4")]
+    public async Task can_bind_timestamp_with_valid_offset(string offsetDays)
+    {
+        HttpRequestMessage request = null;
+        var handler = new InterceptingHttpMessageHandler((message, _) =>
+        {
+            request = message;
+            var response = new HttpResponseMessage(HttpStatusCode.OK);
+            return Task.FromResult(response);
+        });
+        var client = new HttpClient(handler);
+        using var kernel = new HttpKernel(client: client);
+
+        using var _ = new AssertionScope();
+
+        var code = $$$"""
+            POST https://api.example.com/comments
+            
+            {
+                "created_at" : "{{$timestamp {{{offsetDays}}} d}}"
+            }
+            """;
+
+        var result = await kernel.SendAsync(new SubmitCode(code));
+        result.Events.Should().NotContainErrors();
+
+        var bodyAsString = await request.Content.ReadAsStringAsync();
+        var unixValueSubstring = bodyAsString.Split(":").Last().Trim().Substring(1);
+        var unixValueString = unixValueSubstring.Substring(0, unixValueSubstring.IndexOf("\""));
+        var unixValue = DateTimeOffset.FromUnixTimeSeconds(long.Parse(unixValueString));
+        var offsetDaysInteger = int.Parse(offsetDays);
+        var dateTimeOffset = DateTimeOffset.UtcNow.AddDays(offsetDaysInteger);
+        unixValue.Should().BeCloseTo(dateTimeOffset, TimeSpan.FromSeconds(10));
+    }
+
+    [Fact]
+    public async Task cant_bind_timestamp_offset_without_option()
```

Comment:
This test is failing. It seems redundant based on what the `can_bind_timestamp_with_valid_offset` is doing. I think you can either delete it or add `d` after `-1` below.